app/controllers/charge_controller.js

```diff
@@ -238,62 +221,55 @@ module.exports = {
 
-    AWSXRay.captureAsyncFunc('chargeValidator_verify', function (subSegment) {
-      logging.authChargePost(authUrl)
-      AWSXRay.captureAsyncFunc('Charge_patch', function (subSegment) {
-        Charge(req.headers[CORRELATION_HEADER]).patch(req.chargeId, 'replace', 'email', req.body.payerEmail, subSegment)
-          .then(() => {
-            subSegment.close()
-            const startTime = new Date()
-            const correlationId = req.headers[CORRELATION_HEADER] || ''
-            baseClient.post(authUrl, {
-              data: normalise.apiPayload(_.merge(req, convertedPayload), 'visa'),
-              correlationId: correlationId
-            }, (data, json) => {
-              logger.info('[%s] - %s to %s ended - total time %dms', correlationId, 'POST', authUrl, new Date() - startTime)
-              switch (json.statusCode) {
-                case 202:
-                case 409:
-                  logging.failedChargePost(409, authUrl)
-                  redirect(res).toAuthWaiting(req.chargeId)
-                  break
-                case 200:
-                  if (_.get(data, 'status') === State.AUTH_3DS_REQUIRED) {
-                    redirect(res).toAuth3dsRequired(req.chargeId)
-                  } else {
-                    Charge(req.headers[CORRELATION_HEADER])
-                      .capture(req.chargeId)
-                      .then(
-                        () => redirect(res).toReturn(req.chargeId),
-                        err => {
-                          if (err.message === 'CAPTURE_FAILED') return views.display(res, 'CAPTURE_FAILURE', withAnalytics(charge))
-                          views.display(res, 'SYSTEM_ERROR', withAnalytics(
-                            charge,
-                            { returnUrl: routeFor('return', charge.id) }
-                          ))
-                        }
-                      )
-                  }
-                  break
-                case 500:
-                  logging.failedChargePost(409, authUrl)
-                  views.display(res, 'SYSTEM_ERROR', withAnalytics(charge, { returnUrl: routeFor('return', charge.id) }))
-                  break
-                default:
-                  redirect(res).toNew(req.chargeId)
-              }
-            }).on('error', err => {
-              subSegment.close(err)
-              logging.failedChargePostException(err)
-              views.display(res, 'ERROR', withAnalytics(charge))
-            })
-          })
-          .catch(err => {
+    AWSXRay.captureAsyncFunc('Charge_patch', function (subSegment) {
+      Charge(req.headers[CORRELATION_HEADER]).patch(req.chargeId, 'replace', 'email', req.body.payerEmail, subSegment)
+        .then(() => {
+          subSegment.close()
+          const correlationId = req.headers[CORRELATION_HEADER] || ''
+          const payload = normalise.apiPayload(_.merge(req, convertedPayload), 'visa')
```

Comment:
Aaah my bad, look when I was prototyping with PaymentRequest stuff, I hard coded `visa` 😬. Could leave for now perhaps, as  no one is using it anyway